### PR TITLE
[WIP] Add flex attention for qwen2

### PR DESCRIPTION
**Pull request by MekkCyber**
# What does this PR do?
Adding `flex_attention` for Qwen2 models following https://github.com/huggingface/transformers/issues/34809

### CODE REVIEW

File: src/transformers/models/qwen2/modeling_qwen2.py [hunk 1]
```diff
@@ -236,6 +239,115 @@ def repeat_kv(hidden_states: torch.Tensor, n_rep: int) -> torch.Tensor:
...
+        query,
+        key,
+        value,
+        enable_gqa=True,
+        return_lse=output_attentions,
+    )
+    if not output_attentions:
+        return attn_output, None
+    else:
+        return attn_output[0], attn_output[1]
```
Comment (ArthurZucker) [inline src/transformers/models/qwen2/modeling_qwen2.py:340]: Let's do the same as in #34896 for this! 🤗

**Review by ArthurZucker — APPROVED** [review]
Super nice! Thanks for contributing and sorry for being slow!

**Comment by ArthurZucker** [comment]
we need to make sure  CIs are green (make fixup etc) 🤗 do you need help on that ?

**Comment by MekkCyber** [comment]
I just didn't find time to finish it, I will finish it as fast as possible ! thanks 🤗

**Pull request closed without merging**